ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java

```diff
@@ -0,0 +1,215 @@
+   * @param deleteKeys a {@link Set} of collected delete keys
+   * @throws StoreException
+   */
+  private void populateContainerMap(Map<String, Map<String, Long>> validDataSizePerContainer, long referenceTimeInMs,
+      List<MessageInfo> messageInfos, Set<String> deleteKeys) throws StoreException {
+    for (MessageInfo messageInfo : messageInfos) {
+      if (messageInfo.isDeleted()) {
+        // delete record
+        deleteKeys.add(messageInfo.getStoreKey().getID());
+      } else if (!isExpired(messageInfo.getExpirationTimeInMs(), referenceTimeInMs) && !deleteKeys.contains(
+          messageInfo.getStoreKey().getID())) {
+        // put record that is not expired and not deleted
+        updateNestedMapHelper(validDataSizePerContainer, BlobStoreStats.getServiceId(messageInfo.getStoreKey()),
+            BlobStoreStats.getContainerId(messageInfo.getStoreKey()), messageInfo.getSize());
+      }
+    }
+  }
+
+  /**
+   * Process a record in the index and populate the valid size per log segment map.
+   * @param validSizePerLogSegment the valid size per log segment map
+   * @param deleteReferenceTimeInMs the reference time to be used to decide whether or not a blob is deleted
+   * @param expirationReferenceTimeInMs the reference time to be used to decide whether or not a blob is expired
+   * @param messageInfos a list {@link MessageInfo} to be processed
+   * @param lastModifiedTimeInMs the time stamp indicating when the record that is being processed was added
+   * @param logSegmentName the log segment name of the record that is being processed
+   * @param deleteKeys a {@link Set} of collected delete keys
+   * @throws StoreException
+   */
+  private void populateLogSegmentMap(Map<String, Long> validSizePerLogSegment, long deleteReferenceTimeInMs,
+      long expirationReferenceTimeInMs, List<MessageInfo> messageInfos, long lastModifiedTimeInMs, String logSegmentName,
+      Set<String> deleteKeys) throws StoreException {
+    for (MessageInfo messageInfo : messageInfos) {
+      if (messageInfo.isDeleted()) {
+        // delete record, always counted towards valid size
+        updateMapHelper(validSizePerLogSegment, logSegmentName, messageInfo.getSize());
+        if (lastModifiedTimeInMs < deleteReferenceTimeInMs) {
+          // delete that needs to be accounted for
+          deleteKeys.add(messageInfo.getStoreKey().getID());
+        }
+      } else if (!isExpired(messageInfo.getExpirationTimeInMs(), expirationReferenceTimeInMs) && !deleteKeys.contains(
+          messageInfo.getStoreKey().getID())) {
+        // put record that is not expired and not deleted according to the deleteReferenceTime
+        updateMapHelper(validSizePerLogSegment, logSegmentName, messageInfo.getSize());
+      }
+    }
+  }
+
+  private boolean isExpired(long expirationTimeInMs, long referenceTimeInMs) {
+    return expirationTimeInMs < referenceTimeInMs && expirationTimeInMs != Utils.Infinite_Time;
```

Comment:
flip the order of the checks - more intuitive IMO. Something like saying "time is valid and is less than" rather than "time is less than and valid"